### PR TITLE
[CSL 390] android sdk add ability to add filters to autocomplete

### PR DESCRIPTION
**Pull request by crgee1**
- Add support for filtering in autocomplete requests
- Add integration tests for Browse, Search and Autocomplete
- Autocomplete requests only accepts 1 filter at the moment and returns an error if more than 1 is supplied

### CODE REVIEW

File: library/src/main/java/io/constructor/data/remote/ApiPaths.kt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package io.constructor.data.remote
 
 object ApiPaths {
-    const val URL_AUTOCOMPLETE = "autocomplete/{value}"
+    const val URL_AUTOCOMPLETE = "autocomplete/%s"
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/remote/ApiPaths.kt:4]: Can we change to match the other const values with `{term}`?

File: library/src/main/java/io/constructor/core/ConstructorIo.kt [hunk 2]
```diff
@@ -117,13 +117,21 @@ object ConstructorIo {
 
     /**
      * Returns a list of autocomplete suggestions
+     * @param term the term to search for
+     * @param facets additional facets used to refine results
      */
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/core/ConstructorIo.kt:122]: Can we add a comment for `groupId`?

File: library/src/main/java/io/constructor/core/ConstructorIo.kt [hunk 3]
```diff
@@ -117,13 +117,21 @@ object ConstructorIo {
...
-    fun getAutocompleteResults(term: String): Observable<ConstructorData<AutocompleteResponse>> {
-        val params = mutableListOf<Pair<String, String>>()
+    fun getAutocompleteResults(term: String, facets: List<Pair<String, List<String>>>? = null, groupId: Int? = null): Observable<ConstructorData<AutocompleteResponse>> {
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/core/ConstructorIo.kt:123]: Can `facets` be nullable for backwards compatibility?

**Review by rubikzube — CHANGES_REQUESTED** [review]

File: library/src/main/java/io/constructor/data/remote/ApiPaths.kt (hunk 1 above)
Comment (crgee1) [reply library/src/main/java/io/constructor/data/remote/ApiPaths.kt:4]: this one is following the `getSearchResults` and `getBrowseResults` pattern by using `%s`, should I change those as well? @rubikzube

Comment (rubikzube) [reply library/src/main/java/io/constructor/data/remote/ApiPaths.kt:4]: As long as we are consistent one way or the other

**Review by rubikzube — APPROVED** [review]

**Pull request merged**